packages/governance/src/paramManager.js

```diff
@@ -0,0 +1,133 @@
+import { AmountMath, looksLikeBrand } from '@agoric/ertp';
+import { Far } from '@agoric/marshal';
+import { assertKeywordName } from '@agoric/zoe/src/cleanProposal';
+
+/**
+ * @type {{
+ *  AMOUNT: 'amount',
+ *  UNKNOWN: 'unknown',
+ *  BRAND: 'brand',
+ *  INSTANCE: 'instance',
+ *  INSTALLATION: 'installation',
+ *  NAT: 'nat',
+ *  RATIO: 'ratio',
+ *  STRING: 'string',
+ * }}
+ */
+const ParamType = {
+  AMOUNT: 'amount',
+  BRAND: 'brand',
+  INSTANCE: 'instance',
+  INSTALLATION: 'installation',
+  NAT: 'nat',
+  RATIO: 'ratio',
+  STRING: 'string',
+  UNKNOWN: 'unknown',
+};
+harden(ParamType);
+
+const assertType = (type, value, name) => {
+  switch (type) {
+    case ParamType.AMOUNT:
+      // It would be nice to have a clean way to assert something is an amount.
+      assert(
+        AmountMath.isEqual(value, value),
+        X`value for ${name} must be an Amount, was ${value}`,
+      );
+      break;
+    case ParamType.BRAND:
+      assert(
+        looksLikeBrand(value),
+        X`value for ${name} must be a brand, was ${value}`,
+      );
+      break;
+    case ParamType.INSTALLATION:
+      // TODO(3344): add a better assertion once Zoe validates installations
+      assert(
+        typeof value === 'object' && !Object.getOwnPropertyNames(value).length,
+        X`value for ${name} must be an empty object, was ${value}`,
+      );
+      break;
+    case ParamType.INSTANCE:
+      // TODO(3344): add a better assertion once Zoe validates instances
+      assert(
+        typeof value === 'object' && !Object.getOwnPropertyNames(value).length,
+        X`value for ${name} must be an empty object, was ${value}`,
+      );
+      break;
+    case ParamType.NAT:
+      assert.typeof(value, 'bigint');
+      break;
+    case ParamType.RATIO:
+      assertIsRatio(value);
+      break;
+    case ParamType.STRING:
+      assert.typeof(value, 'string');
+      break;
+    case ParamType.UNKNOWN:
+      break;
+    default:
+      assert.fail(X`unknown type guard ${type}`);
+  }
+};
+
+const parse = paramDesc => {
+  const values = {};
+  // manager has an updateFoo() for each Foo param. It will be returned.
+  const manager = {};
+  // getParams() uses describers to generate descriptions of each param.
+  const describers = [];
+
+  paramDesc.forEach(({ name, value, type }) => {
+    assert(
+      !values[name],
+      X`each parameter name must be unique: ${name} duplicated`,
+    );
+    assertType(type, value, name);
+    // we want to create function names like updateFeeRatio(), so we insist that
+    // the name has Keyword-nature.
+    assertKeywordName(name);
+
+    values[name] = { type, value };
+    manager[`update${name}`] = newValue => {
+      assertType(type, newValue, name);
+      values[name].value = newValue;
+    };
+
+    const describer = () => ({
+      name,
+      type,
+      value: values[name].value,
+    });
+    describers.push(describer);
```

Comment:
It seems very strange to have a function per value for making descriptions. Is there a reason we need this? Or could we have a single function that knows how to turn `values` into what should be returned from `getParams`?